Settings: Hide plan purchaser section for non-admins

tyxla opened the pull request:
This PR hides the Site Ownership card non-admin users, as discussed in https://github.com/Automattic/wp-calypso/pull/25725#issuecomment-401390855

To test:
* Checkout this branch.
* Login as the admin of the Jetpack site, owner of the connection and owner of the plan on a site that has at least 1 more connected admin.
* Go to `http://calypso.localhost:3000/settings/manage-connection/:site` where `:site` is your Jetpack site slug.
* Make sure you can see the Site Ownership card.
* Login as a non-admin of the Jetpack site. 
* Verify you can't see the Site Ownership section and the info section below it.

joanrho commented:
@tyxla When I first tested this on an Editor-level user, I could not access Settings at all. It's not visible in the sidebar. When I changed that user's role to Administrator, the Plan Purchaser section disappeared. What role level were you testing on?

Also, would it make sense to hide the entire "Site Ownership" card from non-Admins (which includes both Site Owner and Plan Purchaser)? Or are you breaking that up into two separate PRs?

tyxla commented:
> When I first tested this on an Editor-level user, I could not access Settings at all. It's not visible inhe sidebar. When I changed that user's role to Administrator, the Plan Purchaser section disappeared.

Well, hiding the settings section for non-admins makes sense, because almost all of the settings there require the `manage_options` capability, and non-admins usually don't have it. This, however, seems to reveal a problem: non-admins can't easily reach the disconnect button in Calypso (other than if they go to plugins/jetpack and find the disconnect button there - which is pretty difficult to find). This is definitely material for another PR, but I'd appreciate if anyone from @Automattic/jetpack-design could provide a suggestion of how to fix that and where we could show the disconnect button.

> Also, would it make sense to hide the entire "Site Ownership" card from non-Admins (which includes both Site Owner and Plan Purchaser)? Or are you breaking that up into two separate PRs?

Good point, I've updated the PR to hide the entire card for non-admins. Feel free to have a new look.

ockham commented on client/my-sites/site-settings/manage-connection/site-ownership.jsx:
```diff
@@ -229,7 +228,15 @@ class SiteOwnership extends Component {
+
+		if ( ! canManageOptions ) {
+			return this.renderPlaceholder();
+		}
```
Separate `=== false` and `!` checks are always a bit ugly -- and not too intuitive: I wouldn't know off the top of my head what means what for `canManageOptions`. Given the `return` values, I assume one case is for when we don't have any user and/or site data at all yet (the `!` case), and hence want to show the placeholder.

Can we make that more evident (and get rid of the `=== false` check) by checking for e.g. `! siteId` (or something like that, provided that it's already available in `connect()`), and moving that check up?

I.e. something like

```js
if ( ! siteId ) {
	return this.renderPlaceholder();
}

if ( ! canManageOptions ) {
	return null;
}
```

tyxla replied:
👍 Done in c068b1c460e619cefb2fefc3a7e0572290e653bf.
